activerecord/test/cases/serialized_attribute_test.rb

```diff
@@ -317,6 +318,20 @@ def test_newly_emptied_serialized_hash_is_changed
     assert_equal({}, topic.content)
   end
 
+  if current_adapter?(:Mysql2Adapter)
+    def test_is_not_changed_when_stored_in_mysql_blob
+      value = %w(Fée)
+      model = BinaryField.create!(normal_blob: value, normal_text: value)
+      model.reload
```

Comment:
This PR seems to have broken in-place change detection for cast values, in order to fix it for deserialized ones. If I remove this line, the test passes without the implementation changes from this PR, but fails with them.

Maybe we should be applying binary encoding on assignment, instead of just during change detection?

Reply:
> we should be applying binary encoding on assignment

It indeed sounds like a better solution.